pkg/autoscaling/calculation.go

```diff
@@ -40,14 +44,35 @@ var (
 	MaxScaleInStep uint64 = 1
 )
 
-func calculate(rc *cluster.RaftCluster, strategy *Strategy) []*Plan {
+var (
+	querierInstance Querier
+	once            sync.Once
+)
+
+func getQuerier(cfg *config.PDServerConfig) Querier {
+	once.Do(func() {
+		var err error
+		client, err := promClient.NewClient(promClient.Config{
```

Comment:
What if we modify the `MetricStorage`?

Reply:
It seems that we don't have a notify mechanism for config reloading. Maybe it would be simpler to just create a new client for each request?

Reply:
I think so.